pkg/build/admission/buildpodutil.go

```diff
@@ -0,0 +1,69 @@
+package admission
+
+import (
+	"errors"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+func IsBuildPod(pod *kapi.Pod) bool {
+	return hasBuildAnnotation(pod) && hasBuildEnvVar(pod)
+}
+
+func GetBuildFromPod(pod *kapi.Pod, codec runtime.Codec) (*buildapi.Build, error) {
+	envVar, err := buildEnvVar(pod)
+	if err != nil {
+		return nil, err
+	}
+	build := &buildapi.Build{}
+	err = codec.DecodeInto([]byte(envVar.Value), build)
+	if err != nil {
+		return nil, err
+	}
+	return build, nil
+}
+
+func SetBuildInPod(build *buildapi.Build, pod *kapi.Pod, codec runtime.Codec) error {
+	envVar, err := buildEnvVar(pod)
+	if err != nil {
+		return err
+	}
+	encodedBuild, err := codec.Encode(build)
```

Comment:
> You need a codec that can decode and encode as broadly as possible so that you can handle requests to go to and from any version.

Hmm, not sure that I need that in this case. The encoded build came from the build controller which gets its encoding from the master: https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/run_components.go#L208
I am passing that same codec to the admission control plugin.